src/ray/gcs/tables.cc

```diff
@@ -323,6 +323,42 @@ Status Set<ID, Data>::Remove(const JobID &job_id, const ID &id,
                                        std::move(callback));
 }
 
+template <typename ID, typename Data>
+Status Set<ID, Data>::Replace(const JobID &job_id, const ID &id,
+                              std::shared_ptr<DataT> &old_data,
+                              std::shared_ptr<DataT> &new_data, const WriteCallback &done,
+                              const FailureCallback &failure) {
+  num_removes_++;
+  auto callback = [this, id, new_data, done, failure](const std::string &data) {
+    if (data.length() == 0 && done != nullptr) {
+      (done)(client_, id, *new_data);
+    } else if (data.length() != 0 && failure != nullptr) {
+      (failure)(client_, id);
+    }
+    return true;
+  };
+
+  flatbuffers::FlatBufferBuilder fbb_old;
+  fbb_old.ForceDefaults(true);
+  fbb_old.Finish(Data::Pack(fbb_old, old_data.get()));
+  std::string old_data_string(reinterpret_cast<char *>(fbb_old.GetBufferPointer()),
+                              fbb_old.GetSize());
+  flatbuffers::FlatBufferBuilder fbb_new;
+  fbb_new.ForceDefaults(true);
+  fbb_new.Finish(Data::Pack(fbb_new, new_data.get()));
+  std::string new_data_string(reinterpret_cast<char *>(fbb_new.GetBufferPointer()),
+                              fbb_new.GetSize());
+  flatbuffers::FlatBufferBuilder fbb;
+  auto message = CreateProfileTableData(fbb, fbb.CreateString(old_data_string),
```

Comment:
Does this have something to do with profile table?

Reply:
Good catch! This is totally wrong to use `CreateProfileTableData`. However,  because `ProfileTableData` and `SetReplaceEntryData` have similar structure, there is no error caught.